### PR TITLE
Display package names of external libraries on hover

### PR DESCRIPTION
**Pull request by berberman**
Resolves part of #686

### CODE REVIEW

File: ghcide/src/Development/IDE/Spans/AtPoint.hs [hunk 1]
```diff
@@ -219,11 +225,23 @@ atPoint IdeOptions{} (HAR _ hf _ _ kind) (DKMap dm km) pos = listToMaybe $ point
...
+          m <- nameModule_maybe n
+          let pid = moduleUnitId m
+          conf <- lookupPackage unsafeGlobalDynFlags pid
+          let pkgName = T.pack $ packageNameString conf
+              version = T.pack $ showVersion (packageVersion conf)
+              libName = case sourceLibName conf of
+                Just (PackageName x) -> ":" <> T.pack (unpackFS x)
+                _                    -> ""
+          pure $ " *(" <> pkgName <> "-" <> version <> libName <> ")*"
+
```
Comment (Ailrun) [inline ghcide/src/Development/IDE/Spans/AtPoint.hs:244]: Could you add a test for this? Like whether a function contains its library name or not

**Comment by berberman** [comment]
I couldn't run `findDefinitionAndHoverTests` on my machine at all: 
```
ghcide
  get
    definition
      field in record definition:            FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      field in record construction    #1102: FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      field name used as accessor:           FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      top-level name:                        FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      data constructor record         #1029: FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      data constructor plain:                FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      type constructor                #1028: FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      type constructor external   #717,1028: FAIL (expected: known broken) (0.04s)
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe) (expected failure)
      value external package           #717: FAIL (expected: known broken) (0.04s)
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe) (expected failure)
      plain parameter:                       FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      pattern match name:                    FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      top-level operator               #713: FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      parameter operator:                    FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      name in backticks:                     FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      class in instance declaration   #1027: FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      class in signature              #1027: FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      external class in signature #717,1027: FAIL (expected: known broken) (0.04s)
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe) (expected failure)
      do-notation   bind              #1073: FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      do-notation lookup:                    FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      listcomp   bind                 #1073: FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      listcomp lookup:                       FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      top-level fn 1st clause:               FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      top-level fn 2nd clause         #1030: FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      top-level fn on space           #1002: FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      top-level signature              #767: FAIL (expected: known broken) (0.04s)
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe) (expected failure)
      inner     signature              #767: FAIL (expected: known broken) (0.04s)
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe) (expected failure)
      Imported symbol:                       FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      Imported symbol (reexported):          FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
    hover
      field in record definition:            FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      field in record construction    #1102: FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      field name used as accessor:           FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      top-level name:                        FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      data constructor record         #1029: FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      data constructor plain:                FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      type constructor                #1028: FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      type constructor external   #717,1028: FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      value external package           #717: FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      plain parameter:                       FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      pattern match name:                    FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      top-level operator               #713: FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      parameter operator:                    FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      name in backticks:                     FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      class in instance declaration   #1027: FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      class in signature              #1027: FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      external class in signature #717,1027: FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      do-notation   bind              #1073: FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      do-notation lookup:                    FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      listcomp   bind                 #1073: FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      listcomp lookup:                       FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      top-level fn 1st clause:               FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      top-level fn 2nd clause         #1030: FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      top-level fn on space           #1002: FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      documentation                   #1129: FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      kind of Either                  #1017: FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      kind of Int                     #1017: FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      kind of (* -> *) type variable  #1017: FAIL (expected: known broken) (0.03s)
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe) (expected failure)
      literal Int  in hover info      #1016: FAIL (expected: known broken) (0.04s)
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe) (expected failure)
      literal Char in hover info      #1016: FAIL (expected: known broken) (0.04s)
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe) (expected failure)
      literal Text in hover info      #1016: FAIL (expected: known broken) (0.04s)
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe) (expected failure)
      literal List in hover info      #1016: FAIL (expected: known broken) (0.04s)
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe) (expected failure)
      type constraint in hover info   #1012: FAIL (expected: known broken) (0.05s)
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe) (expected failure)
      top-level signature              #767: FAIL (expected: known broken) (0.07s)
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe) (expected failure)
      inner     signature              #767: FAIL (expected: known broken) (0.04s)
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe) (expected failure)
      hole without internal name       #831: FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      Imported symbol:                       FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      Imported symbol (reexported):          FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      TH Splice Hover:                       FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
    Does GotoHover.hs compile:               OK (0.66s)
    type-definition
      Saturated data con:                    FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)
      Polymorphic variable:                  FAIL
        Exception: fd:24: hPutBuf: resource vanished (Broken pipe)

56 out of 70 tests failed (3.95s)
```
I have no idea about those test suites...

**Comment by pepeiborra** [comment]
"resource vanished - broken pipe" means that HLS stopped running. Either it was killed, it crashed, or it decided to exit on its own. You will need to investigate why by looking at the log output with LSP_TEST_LOG_STDERR=1

**Comment by berberman** [comment]
Server crashed with "[ERROR] Fatal error in server thread: SQLite3 returned ErrorBusy while attempting to perform step: database is locked".

I also found the error gone if I change
https://github.com/haskell/haskell-language-server/blob/28c9f0e60ee7e0c23d55ef90c69ac2971f042ecf/ghcide/test/exe/Main.hs#L5155-L5158
to
```haskell
runWithExtraFiles :: FilePath -> (FilePath -> Session a) -> IO a
runWithExtraFiles prefix s = do
  let dir = "SOME_FIXED_PATH"
  copyTestDataFiles dir prefix
  runInDir dir (s dir)
```

**Comment by pepeiborra** [comment]
#1430

**Review by wz1000 — APPROVED** [review]

**Pull request merged**